core/block.js

```diff
@@ -1477,6 +1480,23 @@ Blockly.Block.newFieldTextInputFromJson_ = function(options) {
   return field;
 };
 
+/**
```

Comment:
Move this into your new field file.  I just filed https://github.com/google/blockly/issues/1486 with an explanation + plans to do the same for the default field types.

Reply:
If I do that, does this file need to be updated to actually use that function? Or is that done automatically somehow?

Reply:
You would change line 1346 (which you just added) to call the new function.